.eslintrc.json

```diff
@@ -1,51 +1,28 @@
 {
-    "extends": [
-      "eslint:recommended",
-      "plugin:@typescript-eslint/eslint-recommended",
-      "plugin:@typescript-eslint/recommended",
-      "standard"
-    ],
-    "parser": "@typescript-eslint/parser",
-    "plugins": ["@typescript-eslint"],
-    "env": { "node": true },
-    "parserOptions": {
-      "ecmaVersion": 6,
-      "sourceType": "module",
-      "project": "./tsconfig.eslint.json",
-      "createDefaultProgram": true
-    },
-    "rules": {
-      "no-console": "off",
-      "@typescript-eslint/indent": ["error", 2],
-      "semi": ["error", "never"],
-      "import/export": "off" // this errors on multiple exports (overload interfaces)
-    },
-    "overrides": [
-      {
-        "files": ["*.d.ts","*.test-d.ts"],
-        "rules": {
-          "no-use-before-define": "off",
-          "no-redeclare": "off",
-          "@typescript-eslint/no-explicit-any": "off"
-        }
+  "plugins": ["@typescript-eslint"],
+  "extends": ["eslint:recommended", "standard"],
+  "overrides": [
+    {
+      "files": ["types/*.test-d.ts", "types/*.d.ts"],
+      "parser": "@typescript-eslint/parser",
+      "parserOptions": {
+        "project": ["./tsconfig.eslint.json"]
       },
-      {
-        "files": ["*.test-d.ts"],
-        "rules": {
-          "@typescript-eslint/no-var-requires": "off",
-          "no-unused-vars": "off",
-          "n/handle-callback-err": "off",
-          "@typescript-eslint/no-empty-function": "off",
-          "@typescript-eslint/explicit-function-return-type": "off",
-          "@typescript-eslint/no-unused-vars": "off",
-          "@typescript-eslint/no-non-null-assertion": "off",
-          "@typescript-eslint/no-misused-promises": ["error", {
-            "checksVoidReturn": false
-          }]
-        },
-        "globals": {
-          "NodeJS": "readonly"
-        }
+      "extends": [
+        "plugin:@typescript-eslint/recommended",
+        "plugin:@typescript-eslint/recommended-requiring-type-checking"
+      ],
+      "rules": {
+        "no-use-before-define": "off",
+        "@typescript-eslint/no-var-requires": "off",
+        "@typescript-eslint/no-explicit-any": "off",
+        "@typescript-eslint/no-floating-promises": "off",
```

Comment:
why we replaced @typescript-eslint/no-misused-promises with this? they are different

Reply:
Good catch, I thought it was unnecessarily disabling the rule instead of enabling it